lib/handler/cache-revalidation-handler.js

```diff
@@ -19,22 +19,30 @@ const DecoratorHandler = require('../handler/decorator-handler')
  */
 class CacheRevalidationHandler extends DecoratorHandler {
   #successful = false
+
   /**
    * @type {((boolean) => void) | null}
    */
   #callback
+
   /**
    * @type {(import('../../types/dispatcher.d.ts').default.DispatchHandlers)}
    */
   #handler
 
+  /**
+   * @type {boolean}
+   */
+  #allowErrorStatusCodes
+
   #abort
 
   /**
    * @param {(boolean) => void} callback Function to call if the cached value is valid
    * @param {import('../../types/dispatcher.d.ts').default.DispatchHandlers} handler
+   * @param {boolean} allowErrorStatusCodes
    */
-  constructor (callback, handler) {
+  constructor (callback, handler, allowErrorStatusCodes) {
```

Comment:
Why is this necessary?

Reply:
For `stale-if-error`, error status codes when revalidating mean to use the stale response